rfcs/0148-taskcluster-worker-location.md

```diff
@@ -0,0 +1,48 @@
+# RFC 148 - TASKCLUSTER_WORKER_LOCATION
+* Comments: [#148](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/148)
+* Proposed by: @djmitche
+
+# Summary
+
+This proposal defines an environment variable, `TASKCLUSTER_WORKER_LOCATION`, containing this information in an easily-digested format.
+
+## Motivation
+
+Workers often need information about where they are running, for example to select nearby mirrors or other resources.
+
+Taskcluster does not currently provide this in a documented, supported fashion.
+
+# Details
+
+This proposes to introduce `$TASKCLUSTER_WORKER_LOCATION` as an environment variable available within tasks run by docker-worker and generic-worker.
+
+The content has the format `<cloud>/<details>`.
+The `<cloud>` portion will correspond to the provider types used within worker-manager (e.g., `google`, `amazon`).
+The `<details>` portion is specific to the cloud.
+Tasks not executed in any cloud (e.g., static or standalone workers) can define arbitrary locations.
+Examples:
+* `amazon/us-east-1b`
+* `google/us-east1-c`
+* `onprem/pdx1`
+
+Workers will always supply a value in this variable, falling back to `unknown/unknown` if no value is known.
```

Comment:
I wonder if it makes more sense to leave `TASKCLUSTER_WORKER_LOCATION` unset if the value isn't known. This way, there's no magic-string parsing to detect unknown worker locations? :thinking:

Reply:
That makes sense, too.  I added this in anticipation of avoiding errors in the dark corners of CI infra that this sort of thing usually collects (e.g., `build-linux.sh`).  But, it's not like 1 in 100 workers will miss this information, so any issues wouldn't be intermittent and would be easy to fix.